test/unittests/execute_floating_point_test.cpp

```diff
@@ -363,6 +363,54 @@ TYPED_TEST(execute_floating_point_types, add)
     EXPECT_THAT(exec(TypeParam{0x0.287p2}, TypeParam{0x1.FFp4}), Result(TypeParam{0x1.048Ep5}));
 }
 
+TEST(execute_floating_point, f64_promote_f32)
+{
+    /* wat2wasm
+    (func (param f32) (result f64)
+      local.get 0
+      f64.promote_f32
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017d017c030201000a070105002000bb0b");
+    auto instance = instantiate(parse(wasm));
+
+    const std::pair<float, double> test_cases[] = {
+        {0.0f, 0.0},
+        {-0.0f, -0.0},
+        {1.0f, 1.0},
+        {-1.0f, -1.0},
+        {FP32::Limits::lowest(), double{FP32::Limits::lowest()}},
+        {FP32::Limits::max(), double{FP32::Limits::max()}},
+        {FP32::Limits::min(), double{FP32::Limits::min()}},
+        {FP32::Limits::denorm_min(), double{FP32::Limits::denorm_min()}},
+        {FP32::Limits::infinity(), FP64::Limits::infinity()},
+        {-FP32::Limits::infinity(), -FP64::Limits::infinity()},
+
+        // The canonical NaN must result in canonical NaN (only the top bit set).
+        {FP32::nan(FP32::canon), FP64::nan(FP64::canon)},
+        {-FP32::nan(FP32::canon), -FP64::nan(FP64::canon)},
+    };
+
+    for (const auto& [arg, expected] : test_cases)
+    {
+        EXPECT_THAT(execute(*instance, 0, {arg}), Result(expected)) << arg << " -> " << expected;
+    }
+
+    // An arithmetic NaN must result in any arithmetic NaN (payload >= canonical payload).
+    // The spec allows the result to be the canonical NaN, but in the current implementation it
+    // is not the case and the check is more restrictive than the spec requires.
+    const auto res1 = execute(*instance, 0, {FP32::nan(FP32::canon + 1)});
```

Comment:
Add also `-FP32::nan(FP32::canon + 1)` and `-FP32::nan(1)` cases

Reply:
Done.